package-res/resources/web/pentaho/type/_doc/IConfigurationService.jsdoc

```diff
@@ -0,0 +1,137 @@
+/*!
+ * Copyright 2010 - 2016 Pentaho Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * The `IConfigurationService` interface describes a service
+ * that manages configurations of [value types]{@link pentaho.type.Value.Type}
+ * of the Pentaho Client Metadata Model.
+ *
+ * The [add]{@link pentaho.type.IConfigurationService#add} method,
+ * is used to register a [spec.ITypeConfiguration]{@link pentaho.type.spec.ITypeConfiguration}
+ * to the service.
+ *
+ * The [select]{@link pentaho.type.IConfigurationService#select} method,
+ * selects the rules of added type configurations that apply to
+ * a given type and context variables,
+ * sorts these by specificity,
+ * merges the rules' type specifications and, finally,
+ * returns the result.
```

Comment:
The [select]{@link pentaho.type.IConfigurationService#select} method
selects the rules of added type configurations that apply to a given type and context variables,
sorts these by specificity, then merges the rules' type specifications and returns the result.